### PR TITLE
flush every chunk only for chunked responses

### PR DESCRIPTION
**Pull request by yanns**
let the servlet container buffer the response.
Based on feedback from @jroper:
https://github.com/yanns/play2-war-plugin/commit/b580b8c9000526aff2d1c6ab07443123fe109177#commitcomment-6312809
https://github.com/yanns/play2-war-plugin/commit/b580b8c9000526aff2d1c6ab07443123fe109177#commitcomment-6312832

### CODE REVIEW

**Comment by dlecan** [comment]
Good work !

File: project-code/core/servlet31/src/main/scala/play/core/server/servlet31/RequestHandler31.scala [hunk 1]
```diff
@@ -223,7 +229,7 @@ class Play2Servlet31RequestHandler(servletRequest: HttpServletRequest)
 
         setHeaders(headers, httpResponse)
 
-        val chunked = headers.exists { case (key, value) => key == HeaderNames.TRANSFER_ENCODING && value == HttpProtocol.CHUNKED }
+        chunked = headers.exists { case (key, value) => key == HeaderNames.TRANSFER_ENCODING && value == HttpProtocol.CHUNKED }
```
Comment (jroper) [inline project-code/core/servlet31/src/main/scala/play/core/server/servlet31/RequestHandler31.scala]: `key.equalsIgnoreCase(...)` would be technically correct here.  Though we don't do it in Play... we probably should.  I think this is going to be fixed by a PR in the works that converts the headers map to be case insensitive.

Comment (yanns) [inline project-code/core/servlet31/src/main/scala/play/core/server/servlet31/RequestHandler31.scala]: thanks, added a commit for this.

**Comment by yanns** [comment]
@dlecan can you merge this pull or do you need more info/time?

**Pull request merged**